src/deck/drivers/src/aideck.c

```diff
@@ -70,15 +171,13 @@ static void NinaTask(void *param)
 
     // Read out the byte the NINA sends and immediately send it to the console.
     uint8_t byte;
-    while (1)
-    {
-        if (uart2GetDataWithDefaultTimeout(&byte) == true)
-        {
-            consolePutchar(byte);
+    while (1) {
+        if (uart2GetDataWithDefaultTimeout(&byte) == true) {
+            ninalinkReceive(byte);
         }
+        vTaskDelay(M2T(10));
```

Comment:
This delay should not be needed and might make it loose packet. Since "uart2GetData" blocks, it will take care of limiting the cpu usage of the task.